src/Kestrel.Core/Internal/Http2/HPack/DynamicTable.cs

```diff
@@ -37,33 +39,48 @@ public DynamicTable(int maxSize)
             }
         }
 
-        public void Insert(string name, string value)
+        public void Insert(Span<byte> name, Span<byte> value)
         {
-            var entrySize = name.Length + value.Length + 32;
-            EnsureSize(_maxSize - entrySize);
+            var entryLength = (name.Length + value.Length + 32);
+            EnsureSize(_maxSize - entryLength);
 
-            if (_maxSize < entrySize)
+            if (entryLength > _maxSize)
             {
-                throw new InvalidOperationException($"Unable to add entry of size {entrySize} to dynamic table of size {_maxSize}.");
+                // http://httpwg.org/specs/rfc7541.html#rfc.section.4.4
+                // It is not an error to attempt to add an entry that is larger than the maximum size;
+                // an attempt to add an entry larger than the maximum size causes the table to be emptied
+                // of all existing entries and results in an empty table.
+                return;
             }
 
-            _buffer[_insertIndex] = new HeaderField(name, value);
+            var entry = new HeaderField(name, value);
+            _buffer[_insertIndex] = entry;
             _insertIndex = (_insertIndex + 1) % _buffer.Length;
-            _size += entrySize;
+            _size += entry.Length;
             _count++;
         }
 
         public void Resize(int maxSize)
         {
             _maxSize = maxSize;
-            EnsureSize(_maxSize);
+
+            if (maxSize > _maxSize)
```

Comment:
Isn't this condition always false b/c of `_maxSize = maxSize` above?

Reply:
Lol yeah.